src/JsonSchema/Metadata/Property/Factory/SchemaPropertyMetadataFactory.php

```diff
@@ -255,16 +262,25 @@ private function getClassType(?string $className, bool $nullable, ?bool $readabl
             ];
         }
 
-        if (true !== $readableLink && $this->isResourceClass($className)) {
+        if (true !== $readableLink && $isResourceClass) {
             return [
                 'type' => 'string',
                 'format' => 'iri-reference',
                 'example' => 'https://example.com/',
             ];
         }
 
-        // TODO: add propertyNameCollectionFactory and recurse to find the underlying schema? Right now SchemaFactory does the job so we don't compute anything here.
-        return ['type' => Schema::UNKNOWN_TYPE];
+        if (!$this->propertyInfo) {
+            return ['type' => 'object'];
+        }
+
+        $properties = $this->propertyInfo->getProperties($className, ['serializer_groups' => null]);
+        $propertiesSchema = [];
+        foreach ($properties as $property) {
+            $propertiesSchema[$property] = $this->create($className, $property)->getSchema();
```

Comment:
```suggestion
            $propertiesSchema[$property] = $this->create($className, $property)->getSchema();
```

we need to split the schema part in a function

Reply:
@soyuka If I do that, the problem is the signature, which looks like this:

```php
 public function createSchema(array $types, ApiProperty $propertyMetadata, ?bool $link, $propertySchema, string $resourceClass, string $property) : ApiProperty
 ```
 In this context, I don't have access to all these things